XIPs/xip-31-message-history.md

```diff
@@ -0,0 +1,435 @@
+        // Load the file from the path, spin up a temporary client with the account keys, download all messages, and create a new Message Backup File
+        // Technically we could simplify this to just load the messages directly into the database without the intermediate file, but I like the idea of reducing surface area and re-using as much code as possible from the remote option
+        ...
+    }
+}
+
+pub struct MessageHistoryBackupRequest {
+    pub requestId: String,
+    pub verificationPin: i16, // A four digit PIN that can be displayed in both the Backup Requester app and the Backup Provider app to ensure the user is responding to the correct backup request
+    pub backupStorageProviderUploadUrl: String,
+    pub status: BackupRequestStatus
+    ...
+}
+
+pub struct MessageHistoryBackupResponse {
+    pub requestId: String,
+    pub verificationPin: i16,
+    backupUrl: String,
+    encryptionKey: Vec<u8>,
+    nonce: Vec<u8>,
+    salt: Vec<u8>,
+    backupFileHash: Vec<u8>,
+    expirationTimeNs: u8
+}
+
+pub enum BackupRequestStatus {
+    Pending,
+    Expired,
+    Applied,
+    Failed
+}
+
+pub enum BackupApplyError {
+    ResponseNotFound,
+    DecryptionFailed,
+    ValidationFailed,
+}
+
+pub enum BackupCreateError {
+    RequestNotFound,
+    InvalidRequest,
+    NetworkError,
+    EncryptionError
+}
+```
+
+## Rationale
+
+There are many other potential solutions to message portability listed below. I am proposing this particular solution because it solves the key user problems stated above with the minimal amount of developer friction. Many of the alternatives require substantial cooperation from app developers, who may care more about shipping new features than supporting competing apps. Developers have strong incentives to receive backups but limited incentives to provide them.
+
+### Alternatives considered
+
+#### Use the XMTP Network as a Backup Storage Provider
+
+Instead of storing backups in a remotely hosted file and serving via HTTP, messages would be divided into chunks and stored on the XMTP network on a special topic only known to the participants in the backup message exchange.
+
+##### XMTP Network Advantages
+
+- No new infrastructure required. Everything stays on the XMTP network
+
+##### XMTP Network Disadvantages
+
+- The XMTP network was not designed for storage of large files. The network currently has a 1MB per message limit so backups would be divided across potentially hundreds of chunks.
+- Costly as the network decentralizes and fees are added to the network. Given that alternative storage providers can be used in the current proposal, the benefits to decentralization are limited.
+
+I could get on board with this proposal if others felt strongly that we shouldn't create new infrastructure. While backups are an odd fit for the XMTP transport network, it would work in the short term and be easier to bootstrap.
```

Comment:
OK to leave this in? Do we want to reword?

Reply:
yes, i think it can be removed. i'll do that in a commit.